### ISSUE TITLE
Crash in Kokkos::Initialize from #4682

### ISSUE THREAD

**Issue by stanmoore1** [kokkos/kokkos#5596]
After updating to 3.7, we are seeing a crash in `Kokkos::Initialize` in LAMMPS with a `DEBUG` build and shared libs. I bisected the issue to https://github.com/kokkos/kokkos/pull/4682, which matches the error messages. If we use a `RELEASE` build or don't enable shared libs, then the code runs fine.

```
what():  Desul::Error: init_lock_arrays_cuda: post init kernel error(cudaErrorLaunchFailure): unspecified launch failure

***FAULT Warp Illegal Address
+++ID CUDA count 832 grid 3 block (0,0,0) thread (0,0,0) dev0/sm0/warp2/lane0
+++STACK
desul::(anonymousnamespace)::init_lock_arrays_cuda_kernel<<<(512,1,1),(256,1,1)>>>()@/lammps_master/lib/kokkos/tpls/desul/src/Lock_Array_CUDA.cpp:32

========= Invalid __global__ write of size 4
=========     at 0x000000d0 in /lammps_master/lib/kokkos/tpls/desul/src/Lock_Array_CUDA.cpp:31:desul::_GLOBAL__N__50_tmpxft_0001a4f7_00000000_7_Lock_Array_CUDA_cpp1_ii_4d2bf3a0::init_lock_arrays_cuda_kernel(void)
=========     by thread (238,0,0) in block (80,0,0)
=========     Address 0x000143b8 is out of bounds
=========     Saved host backtrace up to driver entry point at kernel launch time
```

**Comment by stanmoore1** [comment]
@bathmatt

**Comment by stanmoore1** [comment]
@masterleinad any update on this? Do we need to revert #4682?

**Comment by masterleinad** [comment]
> @masterleinad any update on this? Do we need to revert https://github.com/kokkos/kokkos/pull/4682?

I don't really understand the issue, yet, and reverting seems to be the best option at the moment but we are still debating how to deal with it.

**Comment by masterleinad** [comment]
Reproducer using `LAMMPS`:
```
git clone https://github.com/lammps/lammps.git
cd lammps
mkdir build
cd build
cmake -C ../cmake/presets/basic.cmake -C ../cmake/presets/kokkos-cuda.cmake -D BUILD_SHARED_LIBS=on -DPKG_KOKKOS=ON -DCMAKE_BUILD_TYPE=DEBUG -DKokkos_ARCH_VOLTA70=ON -DKokkos_ARCH_PASCAL60=OFF ../cmake/
make -j16
cd ../bench
../build/lmp -in in.lj -k on g 1 -sf kk -pk kokkos
```

**Comment by PhilMiller** [comment]
Removed the 'Patch Release' label as this already shipped in 3.7.01, and I don't want it to pollute searches for forward-looking work.

**Comment by PhilMiller** [comment]
Re-adding patch release label, to reflect potential intention to cherry-pick the new fix over to 3.7.02 in place of the old one, per https://github.com/kokkos/kokkos/pull/5709#issuecomment-1360045004

**Comment by stanmoore1** [comment]
Fixed by https://github.com/lammps/lammps/pull/3680
